pkg/cloudprovider/providers/gce/gce_clusterid.go

```diff
@@ -105,7 +105,7 @@ func (gce *GCECloud) watchClusterID() {
 	var controller cache.Controller
 	gce.ClusterID.store, controller = cache.NewInformer(newSingleObjectListerWatcher(listerWatcher, UIDConfigMapName), &v1.ConfigMap{}, updateFuncFrequency, mapEventHandler)
 
-	controller.Run(nil)
+	controller.Run(stop)
```

Comment:
Wondering if it would be good to this in a separate PR. I believe we have other such fixes which need to be made. Might be better to not conflate the Interface change with actual fixes which may have unexpected side effects.

Reply:
I can at least do separate commit

Reply:
done, PTAL :)